Packs/OpenAI/Integrations/OpenAIGPT/OpenAIGPT.py

```diff
@@ -0,0 +1,454 @@
+                                                                 }
+                                                                 )
+    return CommandResults(
+        outputs_prefix='OpenAIGPT.Conversation',
+        outputs=conversation_step,
+        replace_existing=reset_conversation_history,
+        readable_output=readable_output
+    ), response
+
+
+def check_email_headers_command(client: OpenAiClient, args: dict[str, Any]) -> CommandResults:
+    return check_email_part(EmailParts.HEADERS, client, args)
+
+
+def check_email_body_command(client: OpenAiClient, args: dict[str, Any]) -> CommandResults:
+    return check_email_part(EmailParts.BODY, client, args)
+
+
+def create_soc_email_template_command(client: OpenAiClient, args: dict[str, Any]) -> CommandResults:
+    additional_instructions = f'Additional instructions: {args.get(ArgAndParamNames.ADDITIONAL_INSTRUCTIONS)}\n'\
+        if args.get(ArgAndParamNames.ADDITIONAL_INSTRUCTIONS, "") else ''
+    create_soc_email_template_message = CREATE_SOC_EMAIL_TEMPLATE_PROMPT.format(additional_instructions)
+    args.update({ArgAndParamNames.MESSAGE: create_soc_email_template_message})
+    send_message_command_results, response = send_message_command(client, args)
+    # Setting the SOCEmailTemplate context prior to returning the 'send-message-command' results
+    # and setting the entire conversation in the context.
+    return_results(
+        CommandResults(
+            outputs_prefix='OpenAIGPT.SocEmailTemplate',
+            outputs={'Response': response},
+            replace_existing=True
+        )
+    )
+    return send_message_command_results
+
+
+''' MAIN FUNCTION '''
+
+
+def main() -> None:  # pragma: no cover
+    """main function, parses params and runs command functions
+
+    :return:
+    :rtype:
+    """
+
+    params = demisto.params()
+    args = demisto.args()
+    command = demisto.command()
+
+    api_key = params.get('api_key')
+    # If a model name was provided within the free text box, it will override the selected one from the model selection box.
+    # The provided model will be tested for compatability within the test module.
+    model = params.get('model-freetext') if params.get('model-freetext') else params.get('model-select')
+
+    # Using instance params for model configuration, if command args were not provided.
+    if not args.get(ArgAndParamNames.MAX_TOKENS, None) and params.get(ArgAndParamNames.MAX_TOKENS, None):
+        args[ArgAndParamNames.MAX_TOKENS] = params.get(ArgAndParamNames.MAX_TOKENS)
+    if not args.get(ArgAndParamNames.TEMPERATURE, None) and params.get(ArgAndParamNames.TEMPERATURE, None):
+        args[ArgAndParamNames.TEMPERATURE] = params.get(ArgAndParamNames.TEMPERATURE)
+    if not args.get(ArgAndParamNames.TOP_P, None) and params.get(ArgAndParamNames.TOP_P, False):
+        args[ArgAndParamNames.TOP_P] = params.get(ArgAndParamNames.TOP_P)
```

Comment:
I would extract this block to a method named setup_args or something like that with the comment above the block as doc string.

Reply:
done